src/Storages/ObjectStorage/StorageObjectStorage.cpp

```diff
@@ -52,15 +72,22 @@ StorageObjectStorage::StorageObjectStorage(
     , log(getLogger(fmt::format("Storage{}({})", configuration->getEngineName(), table_id_.getFullTableName())))
 {
     ColumnsDescription columns{columns_};
-    resolveSchemaAndFormat(columns, configuration->format, object_storage, configuration, format_settings, context);
+
+    std::string sample_path;
+    resolveSchemaAndFormat(columns, configuration->format, object_storage, configuration, format_settings, sample_path, context);
     configuration->check(context);
 
     StorageInMemoryMetadata metadata;
     metadata.setColumns(columns);
     metadata.setConstraints(constraints_);
     metadata.setComment(comment);
 
-    setVirtuals(VirtualColumnUtils::getVirtualsForFileLikeStorage(metadata.getColumns()));
+    if (sample_path.empty() && context->getSettings().use_hive_partitioning)
+        sample_path = getPathSample(metadata, context);
+    else if (!context->getSettings().use_hive_partitioning)
+        sample_path = "";
```

Comment:
If we will move checking `use_hive_partitioning` to `getVirtualsForFileLikeStorage`, we can avoid the code of `else if` branch.